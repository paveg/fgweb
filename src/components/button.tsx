
type Props = {
  apperance: 'primary' | 'secondary' | 'error'
  onClick?: () => void,
}

const primaryColor = 'border-green-500 bg-green-500 hover:bg-green-600'
const secondaryColor = 'border-yellow-500 bg-yellow-500 hover:bg-yellow-600'
const errorColor = 'border-red-500 bg-red-500 hover:bg-red-600'

export const Button: React.FC<Props> = ({ apperance, onClick, children }): JSX.Element => {
  let color: string
  switch (apperance) {
    case 'primary':
      color = primaryColor
      break
    case "secondary":
      color = secondaryColor
      break
    case 'error':
      color = errorColor
      break
    default:
      color = primaryColor
      break
  }

  return <button
    type="button"
    onClick={onClick}
    className={`${color} border text-white rounded-md px-4 py-2 m-2 transition duration-500 ease select-none focus:outline-none focus:shadow-outline`}>
    {children}
  </button>
}
