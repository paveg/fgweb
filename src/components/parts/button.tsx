
type Props = {
  apperance: 'primary' | 'secondary' | 'error' | 'neutral'
  onClick?: () => void,
}

const primaryColor = 'border-green-500 bg-green-500 hover:bg-green-600 text-white'
const secondaryColor = 'border-yellow-500 bg-yellow-500 hover:bg-yellow-600 text-white'
const errorColor = 'border-red-500 bg-red-500 hover:bg-red-600 text-white'
const neutralColor = 'border-gray-200 bg-gray-200 hover:bg-gray-300 text-black-900'

export const Button: React.FC<Props> = ({ apperance, onClick, children }): JSX.Element => {
  let color: string
  switch (apperance) {
    case 'primary':
      color = primaryColor
      break
    case "secondary":
      color = secondaryColor
      break
    case 'error':
      color = errorColor
      break
    case 'neutral':
      color = neutralColor
      break
    default:
      color = primaryColor
      break
  }

  return <button
    type="button"
    onClick={onClick}
    className={`${color} border rounded-md px-4 py-2 m-2 transition duration-500 ease select-none focus:outline-none focus:shadow-outline`}>
    {children}
  </button>
}
