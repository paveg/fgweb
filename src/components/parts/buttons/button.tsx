export type ButtonProps = {
  apperance: 'primary' | 'secondary' | 'error' | 'neutral' | 'tertiary'
  small?: boolean
  onClick?: React.MouseEventHandler
  disabled?: boolean
}

const primaryColor =
  'border-green-500 bg-green-500 hover:bg-green-600 text-white border'
const secondaryColor =
  'border-yellow-500 bg-yellow-500 hover:bg-yellow-600 text-white border'
const errorColor =
  'border-red-500 bg-red-500 hover:bg-red-600 text-white border'
const neutralColor =
  'border-gray-200 bg-gray-200 hover:bg-gray-300 text-black-900 border'
const tertiaryColor = 'hover:bg-gray-200 text-gray-900'

export const Button: React.FC<ButtonProps> = ({
  apperance,
  small,
  onClick,
  disabled,
  children,
}): JSX.Element => {
  let color: string
  switch (apperance) {
    case 'primary':
      color = primaryColor
      break
    case 'secondary':
      color = secondaryColor
      break
    case 'error':
      color = errorColor
      break
    case 'neutral':
      color = neutralColor
      break
    case 'tertiary':
      color = tertiaryColor
      break
    default:
      color = primaryColor
      break
  }
  const sizeClass = small ? 'px-4 py-2 text-xs' : 'px-6 py-3 text-sm'

  return (
    <button
      type="button"
      onClick={(e: React.MouseEvent): void => {
        disabled ? e.preventDefault() : onClick && onClick(e)
      }}
      disabled={disabled && true}
      className={`${color} ${sizeClass} disabled:opacity-60 m-1 rounded-md transition duration-500 ease select-none focus:outline-none focus:shadow-outline`}
    >
      {children}
    </button>
  )
}
