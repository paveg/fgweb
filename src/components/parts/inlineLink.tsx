import Link from 'next/link'

export type InlineLinkProps = {
  href: string
  target?: '_blank'
}

export const InlineLink: React.FC<InlineLinkProps> = ({ href, target, children }): JSX.Element => {
  return (target === '_blank') ?
    <Link href={href}>
      <a className="underline text-blue-600 hover:text-blue-800 visited:text-purple-600 mx-12" target={target} rel="noreferrer noopener">
        {children}
      </a>
    </Link> :
    <Link href={href}>
      <a className="underline text-blue-600 hover:text-blue-800 visited:text-purple-600 mx-12">
        {children}
      </a>
    </Link>
}
