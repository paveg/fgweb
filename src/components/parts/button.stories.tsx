import React from 'react';
import { Story, Meta } from '@storybook/react';

import { Button, ButtonProps } from './button';

export default {
  title: 'Example/Button',
  component: Button,
} as Meta;

const Template: Story<ButtonProps> = (args) => {
  return <Button {...args} >Button</Button>
};

export const Default = Template.bind({});
