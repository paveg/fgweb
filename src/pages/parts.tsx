import React from 'react'
import { Layout } from '../components/layout'
import { Button } from '../components/parts/buttons/button'

export const Parts: React.FC = (): JSX.Element => (
  <Layout pageTitle="Parts">
    <div className="w-full text-center mx-auto">
      <h2>Buttons</h2>
      <Button apperance="primary">Primary</Button>
      <Button apperance="primary" small>
        Small Primary
      </Button>
      <Button apperance="secondary">Secondary</Button>
      <Button apperance="neutral">Neutral</Button>
      <Button
        apperance="error"
        onClick={() => {
          return window.alert('alert opened.')
        }}
      >
        Alert
      </Button>
      <Button apperance="error" disabled>
        Disabled
      </Button>
    </div>
  </Layout>
)

export default Parts
