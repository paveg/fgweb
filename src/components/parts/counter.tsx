import { countState } from '../../states'
import { useRecoilState } from 'recoil'
import { Button } from './button'
import { InlineLink } from './inlineLink'
import React from 'react'

export const Counter = () => {
  const [count, setCount] = useRecoilState(countState)

  return <div className="mt-3">
    <p><InlineLink href="https://github.com/facebookexperimental/Recoil">Recoil</InlineLink> Sample</p><br />
    <span className="ml-3">count: {count}</span><br />
    <Button apperance="primary" onClick={() => setCount((c) => c + 1)}>count up</Button>
  </div>
}
