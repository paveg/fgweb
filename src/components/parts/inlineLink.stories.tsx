import React from 'react'
import { Story, Meta } from '@storybook/react'

import { InlineLink, InlineLinkProps } from './inlineLink'

export default {
  title: 'Example/InlineLink',
  component: InlineLink,
} as Meta

const Template: Story<InlineLinkProps> = (args) => {
  return <InlineLink {...args}>InlineLink</InlineLink>
}

export const Default = Template.bind({})
Default.args = {
  href: '/',
  target: '_blank',
}
