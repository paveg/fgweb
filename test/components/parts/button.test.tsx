import React from 'react'
import { render } from '../../testUtils'
import { Button } from '../../../src/components/parts/button'

describe('Button', () => {
  it('matches snapshot primary', () => {
    const { asFragment } = render(<Button apperance="primary">Primary</Button>, {})
    expect(asFragment()).toMatchSnapshot()
  })
  it('matches snapshot secondary', () => {
    const { asFragment } = render(<Button apperance="secondary">Secondary</Button>, {})
    expect(asFragment()).toMatchSnapshot()
  })
  it('matches snapshot neutral', () => {
    const { asFragment } = render(<Button apperance="neutral">Neutral</Button>, {})
    expect(asFragment()).toMatchSnapshot()
  })
  it('matches snapshot error', () => {
    const { asFragment } = render(<Button apperance="error">Error</Button>, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('matches snapshot small', () => {
    const { asFragment } = render(<Button apperance="primary" small>Disabled</Button>, {})
    expect(asFragment()).toMatchSnapshot()
  })

  it('matches snapshot disabled', () => {
    const { asFragment } = render(<Button apperance="primary" disabled>Disabled</Button>, {})
    expect(asFragment()).toMatchSnapshot()
  })
})
