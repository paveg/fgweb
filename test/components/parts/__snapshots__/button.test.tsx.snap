// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button matches snapshot disabled 1`] = `
<DocumentFragment>
  <button
    class="border-green-500 bg-green-500 hover:bg-green-600 text-white border px-6 py-3 text-sm disabled:opacity-60 m-1 rounded-md transition duration-500 ease select-none focus:outline-none focus:shadow-outline"
    disabled=""
    type="button"
  >
    Disabled
  </button>
</DocumentFragment>
`;

exports[`Button matches snapshot error 1`] = `
<DocumentFragment>
  <button
    class="border-red-500 bg-red-500 hover:bg-red-600 text-white border px-6 py-3 text-sm disabled:opacity-60 m-1 rounded-md transition duration-500 ease select-none focus:outline-none focus:shadow-outline"
    type="button"
  >
    Error
  </button>
</DocumentFragment>
`;

exports[`Button matches snapshot neutral 1`] = `
<DocumentFragment>
  <button
    class="border-gray-200 bg-gray-200 hover:bg-gray-300 text-black-900 border px-6 py-3 text-sm disabled:opacity-60 m-1 rounded-md transition duration-500 ease select-none focus:outline-none focus:shadow-outline"
    type="button"
  >
    Neutral
  </button>
</DocumentFragment>
`;

exports[`Button matches snapshot primary 1`] = `
<DocumentFragment>
  <button
    class="border-green-500 bg-green-500 hover:bg-green-600 text-white border px-6 py-3 text-sm disabled:opacity-60 m-1 rounded-md transition duration-500 ease select-none focus:outline-none focus:shadow-outline"
    type="button"
  >
    Primary
  </button>
</DocumentFragment>
`;

exports[`Button matches snapshot secondary 1`] = `
<DocumentFragment>
  <button
    class="border-yellow-500 bg-yellow-500 hover:bg-yellow-600 text-white border px-6 py-3 text-sm disabled:opacity-60 m-1 rounded-md transition duration-500 ease select-none focus:outline-none focus:shadow-outline"
    type="button"
  >
    Secondary
  </button>
</DocumentFragment>
`;

exports[`Button matches snapshot small 1`] = `
<DocumentFragment>
  <button
    class="border-green-500 bg-green-500 hover:bg-green-600 text-white border px-4 py-2 text-xs disabled:opacity-60 m-1 rounded-md transition duration-500 ease select-none focus:outline-none focus:shadow-outline"
    type="button"
  >
    Disabled
  </button>
</DocumentFragment>
`;
